extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/DataSegmentProvider.java

```diff
@@ -40,4 +41,9 @@ Supplier<ResourceHolder<Segment>> fetchSegment(
       ChannelCounters channelCounters,
       boolean isReindex
   );
+
+  DataSegment fetchDataSegment(
```

Comment:
I donot think these changes should be bundled with this PR.

Reply:
Moved these changes out. Will raise a separate PR.